### PR TITLE
Add blockstore precompiles

### PR DESCRIPTION
**Pull request by lthibault**
⚠️ Stacks on top of #154.  Merge that first.

## 📝 Summary

Adds two precompiles for interacting directly with the datastore (tentatively renamed to `Blockstore`, though this may change).

## 📚 References

---

* [x] I have seen and agree to CONTRIBUTING.md

### CODE REVIEW

**Comment by lthibault** [comment]
Merge conflict is noted.  Will resolve once #154 is merged in.

**Comment by metachris** [comment]
should this PR point to the branch of #154 instead of `main`? not sure

File: core/vm/offchain.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package vm
+
+import (
+	"github.com/ethereum/go-ethereum/suave/offchain"
+)
+
+type System interface {
+	Blocks() offchain.Blockstore
```
Comment (metachris) [inline core/vm/offchain.go:8]: why is this using `Blocks` and `Blockstore`? wouldn't this be kind of a generic store? the name implies it's for blocks 🤔

File: eth/backend.go [hunk 2]
```diff
@@ -319,6 +321,11 @@ func New(stack *node.Node, config *ethconfig.Config) (*Ethereum, error) {
 	stack.RegisterLifecycle(eth)
 	stack.RegisterLifecycle(confidentialStoreEngine)
 
+	// TODO(lthibault):  replace this with a proper CLI flag.
+	if os.Getenv("SUAVE_EXPERIMENTAL_IPFS") != "" {
+		stack.RegisterLifecycle(&offchain.Env{})
```
Comment (metachris) [inline eth/backend.go:326]: ideally add a log statement here so it's clearly visible on startup that this is indeed enabled

File: core/vm/offchain.go (hunk 1 above)
Comment (lthibault) [reply core/vm/offchain.go:8]: Yeah, it's confusing.  I'll change this to something like `Blob`, which is more accurate.

**Comment by lthibault** [comment]
> should this PR point to the branch of #154 instead of `main`? not sure

@metachris I was hoping to merge these sequentially into `main`, to build incrementally.

**Pull request closed without merging**